feat: implement landing page and sidebar UI

Rl0007 opened the pull request:
<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Public landing site/layout, interactive sidebar, hero & features; App Version list and DocType
  * Redesigned Apps UI (Frappe vs Custom), custom app form with branch support, register/delete custom app APIs
  * Custom image preview/build/enqueue, build status endpoint, image tagging and rebuild flow
  * Theme design tokens, input mask plugin, blink animation utility

* **Bug Fixes**
  * Home page routing corrected; success toast color fixed

* **Chores**
  * CI runtimes updated; deployment playbook now performs a fresh repo clone

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Venkateshvenki404224 committed:
wip: implement landing page and sidebar UI

Add public landing page and PenguinUI sidebar navigation.
Still needs browser cache clearing and layout fixes.

Partial implementation of #19

Venkateshvenki404224 committed:
feat: implement Penguin UI sidebar and fix width issue (#19)

Venkateshvenki404224 committed:
fix: update toast colors and replace remove button with icon (#23)

- Fix success toast notification color from red to green
- Replace text "Remove" button with Heroicons trash icon in red

Venkateshvenki404224 committed:
fix: the steps should have less z index than navbar

Venkateshvenki404224 committed:
feat: add custom app section, in the custom app dialog show tabs if the app is public or private, based on this toggle if the github token is required or not.

Venkateshvenki404224 committed:
fix(ui): add separate sections for Frappe Apps and Custom Apps

- Split apps into two distinct sections with headers
- Frappe Apps section displays core apps (frappe === 1) with green gradient selection
- Custom Apps section shows database custom apps and user-added apps
- Updated backend to fetch 'frappe' field from Apps doctype
- Added computed properties to filter apps by type
- Implemented inline error validation for custom app form
- Fixed domain verification to always re-verify on each visit

Venkateshvenki404224 committed:
fix(ci): update Python version to 3.12

Venkateshvenki404224 committed:
fix(ci): update Python version to 3.14

Venkateshvenki404224 committed:
fix(ci): update node version to 24

Venkateshvenki404224 committed:
fix(ui): redesign the summery page

Venkateshvenki404224 committed:
fix: custom image build non type error

Venkateshvenki404224 committed:
chore: removed the schedular to ping server

Venkateshvenki404224 committed:
feat: add DNS record instruction like frappe cloud

Venkateshvenki404224 committed:
feat: improve domain UI and add custom apps management

Rl0007 commented on nano_press/hooks.py:
```diff
@@ -151,13 +151,13 @@
-		],
-	}
-}
+# scheduler_events = {
+# 	"cron": {
+# 		"*/5 * * * *": [
+# 			"nano_press.utils.ansible_runner.ping_server",
+# 		],
+# 	}
+# }
```
Why are we removing this?

Venkateshvenki404224 replied:
Because the ping server function expects ip address or server doc so I didn't have any idea on how to pass argument from hooks.py

Rl0007 commented on nano_press/public/js/script.js:
```diff
@@ -56,16 +64,59 @@
-				});
-				this.customAppForm = { name: '', githubUrl: '', token: '' };
-				this.showCustomAppDialog = false;
+			// Clear all errors
```
Is this comment needed?

Rl0007 commented on nano_press/public/js/script.js:
```diff
@@ -56,16 +64,59 @@
+
+			if (!this.customAppForm.branch) {
+				this.formErrors.branch = 'Branch is required.';
+				hasError = true;
```
Instead of manually checking can we use the HTML validation like this https://github.com/BuildWithHussain/ls_shop/blob/develop/ls_shop/www/account/profile.html

On form submit we collect the data and add additional validations, everything else is handled by HTML form validations

Rl0007 commented on nano_press/templates/components/core/apps.html:
```diff
@@ -19,41 +19,69 @@ <h2 class="text-xl font-semibold mb-6 text-gray-900">Select Apps</h2>
+        <!-- SECTION 2: Custom Apps -->
+        <h3 class="text-sm font-semibold mb-3 mt-6 text-gray-900">Custom Apps</h3>
+
+        <!-- Database custom apps (frappe === 0) -->
```
Can we remove this comment?

Rl0007 commented on nano_press/templates/components/core/apps.html:
```diff
@@ -19,41 +19,69 @@ <h2 class="text-xl font-semibold mb-6 text-gray-900">Select Apps</h2>
         </div>
     </template>
 
+    <!-- "Add Custom App" Button -->
```
Let's remove this and other comments

Rl0007 commented on nano_press/templates/includes/sidebar.html:
```diff
@@ -0,0 +1,98 @@
+{% set full_name = frappe.utils.get_fullname(frappe.session.user) if frappe.utils.get_fullname else frappe.session.user %}
+
+<div x-data="{ showSidebar: false }" class="relative flex w-full flex-col md:flex-row">
+    <!-- This allows screen readers to skip the sidebar and go directly to the main content. -->
```
Let's remove this comments

Rl0007 commented on nano_press/templates/layout.html:
```diff
@@ -38,41 +37,38 @@
+    {% include "templates/includes/sidebar.html" %}
+
+                <div class="overflow-y-auto" x-data="toaster">
+                    <div class="w-full px-6 md:px-10 py-8">
```
Let's cap the width at max-w-7xl.
Reason being on extremely large devices it takes complete width
<img width="1470" height="361" alt="Screenshot 2025-12-27 at 11 04 51 AM" src="https://github.com/user-attachments/assets/c28e9afd-f09e-4f18-b927-6bbb2d5370fa" />

It's better to have it contained.

Rl0007 commented on nano_press/templates/components/landing/features.html:
```diff
@@ -0,0 +1,100 @@
+        </div>
+
+        <div class="grid grid-cols-1 md:grid-cols-3 gap-8">
+            <div class="group p-8 from-gray-50 to-white border border-gray-200 rounded-xl hover:shadow-xl transition-all duration-300 hover:-translate-y-1">
```
We don't need gradient

Venkateshvenki404224 committed:
feat: populate old server list when its already available.

Venkateshvenki404224 committed:
feat(ux): add frappe icon in list apps and add app version for maintaining the frappe versions

Venkateshvenki404224 committed:
fix: linter

Venkateshvenki404224 committed:
chore: rename the placeholder

Venkateshvenki404224 committed:
wip(custom_image): Working on Custom Image

Venkateshvenki404224 committed:
feat: improve domain setup and apps management UI

Venkateshvenki404224 committed:
refactor: add translation support to error messages

Venkateshvenki404224 committed:
fix(security): secure GitHub token storage and improve custom apps management

Security Fixes:
- Change pat_token field from Data to Password for encrypted storage
- Add register_custom_app() API to securely handle token storage
- Tokens no longer stored in localStorage (only reference IDs persist)
- Tokens cleared from client-side immediately after backend POST
- Add delete_custom_app() API to properly remove apps from backend
- Tokens encrypted at rest using Frappe's Password field

Venkateshvenki404224 commented on nano_press/nano_press/doctype/custom_image/custom_image.py:
```diff
@@ -294,134 +184,96 @@ def _send_email_notification(self, status: str):
 
-		custom_image.insert(ignore_permissions=True)
+		if not frappe.db.exists("Apps", app_name):
+			frappe.get_doc(
+				{
+					"doctype": "Apps",
+					"app_name": app_name,
+					"repo_url": custom_app.get("githubUrl", ""),
+					"branch": custom_app.get("branch", "main"),
+					"is_custom": 1,
+					"is_public": 1,
+					"enabled": 1,
+					"order": 999,
+				}
+			).insert(ignore_permissions=True)
```
In the new secure flow, tokens are NO LONGER in the custom_apps payload because they're posted to the backend immediately. The custom_apps array only contains {name, githubUrl, branch, app_reference_id} - no tokens. So this code is actually correct for the new secure flow. Apps with tokens are already in the database from the register_custom_app API call.

Venkateshvenki404224 committed:
fix(custom_image): improve validation, sanitization and performance

- Fix Docker tag sanitization (handle special chars, leading dots/dashes)
- Fix app sorting bug with index alignment
- Optimize deployment vars to use pre-computed apps_json_base64
- Add server existence validation before build
- Validate custom app entries require name and repo_url

Venkateshvenki404224 committed:
fix: improve custom image build and secure custom app

Venkateshvenki404224 committed:
wip: add URL state management for deployment resume

Implement URL-based deployment tracking to preserve progress on page refresh.
When user refreshes during deployment, polling automatically resumes and
continues from current status. Removes blocking image build wait.

Venkateshvenki404224 committed:
feat(deployment): sync custom image build status to site for seamless deployment tracking

This update enables automatic status synchronization between Custom Image builds and Frappe Sites, allowing the frontend to track the entire deployment lifecycle by polling a single DocType.

Venkateshvenki404224 committed:
refactor(ui): improve server details layout and styling

Venkateshvenki404224 committed:
refactor: fix memory leaks, redundant operations, and improve validation

Cleaned up several code quality issues identified during review to improve reliability and user experience.

Fixes:
- Remove duplicate status sync call in custom image build failure handler
- Fix polling interval memory leak by tracking and cleaning up on component destroy
- Ensure summary component visible when resuming from URL parameters
- Add null safety fallback for user servers array to prevent runtime errors
- Remove redundant status update in rebuild endpoint (already handled by sync)
- Upgrade failed image validation from warning to hard block for data integrity
- Add accessibility support for users with reduced motion preferences

Venkateshvenki404224 committed:
chore: add app version fixtures
